types/trusted-types/index.d.ts

```diff
@@ -29,48 +23,57 @@ declare global {
-        private brand: true; // To prevent structural typing.
+    interface TrustedTypePolicyFactory {
+        createPolicy<Options extends TrustedTypePolicyOptions>(
+            policyName: string,
+            policyOptions?: Options,
+        ): TypedPolicy<Options>;
+        isHTML(value: unknown): value is TrustedHTML;
+        isScript(value: unknown): value is TrustedScript;
+        isScriptURL(value: unknown): value is TrustedScriptURL;
+        readonly emptyHTML: TrustedHTML;
+        readonly emptyScript: TrustedScript;
+        getAttributeType(tagName: string, attribute: string, elementNs?: string, attrNs?: string): string | null;
+        getPropertyType(tagName: string, property: string, elementNs?: string): string | null;
+        readonly defaultPolicy: TrustedTypePolicy | null;
     }
 
     interface TrustedTypePolicy {
         readonly name: string;
-        createHTML(input: string): TrustedHTML;
-        createScript(input: string): TrustedScript;
-        createScriptURL(input: string): TrustedScriptURL;
-        createURL(input: string): TrustedURL;
+        createHTML(input: string, ...arguments: any[]): TrustedHTML;
+        createScript(input: string, ...arguments: any[]): TrustedScript;
+        createScriptURL(input: string, ...arguments: any[]): TrustedScriptURL;
     }
 
-    interface TrustedTypePolicyFactory {
-        createPolicy<Keys extends keyof TrustedTypePolicyOptions>(
-            name: string,
-            policyOptions: Pick<TrustedTypePolicyOptions, Keys>,
-            expose?: boolean,
-        ): Pick<TrustedTypePolicy, 'name' | Keys>;
-        getPolicyNames(): string[];
-        isHTML(value: any): value is TrustedHTML;
-        isScript(value: any): value is TrustedScript;
-        isScriptURL(value: any): value is TrustedScriptURL;
-        isURL(value: any): value is TrustedURL;
-        getAttributeType(tagName: string, attrName: string, elemNs?: string, attrNs?: string): string | undefined;
-        getPropertyType(tagName: string, propName: string, elemNs?: string): string | undefined;
-        defaultPolicy?: TrustedTypePolicy;
-        emptyHTML: TrustedHTML;
+    interface TrustedTypePolicyOptions {
+        createHTML?: (input: string, ...arguments: any[]) => string;
+        createScript?: (input: string, ...arguments: any[]) => string;
+        createScriptURL?: (input: string, ...arguments: any[]) => string;
     }
 
+    type FnNames = keyof TrustedTypePolicyOptions;
+    type Args<Options extends TrustedTypePolicyOptions, K extends FnNames> =
+        Parameters<NonNullable<Options[K]>>;
+
+    type FullTypedPolicy<Options extends TrustedTypePolicyOptions> = {
+        name: string,
+        createHTML(...args: Args<Options, 'createHTML'>): TrustedHTML,
+        createScript(...args: Args<Options, 'createScript'>): TrustedScript,
+        createScriptURL(...args: Args<Options, 'createScriptURL'>): TrustedScriptURL,
+    };
```

Comment:
```suggestion
    interface FullTypedPolicy<Options extends TrustedTypePolicyOptions> {
        name: string;
        createHTML(...args: Args<Options, 'createHTML'>): TrustedHTML;
        createScript(...args: Args<Options, 'createScript'>): TrustedScript;
        createScriptURL(...args: Args<Options, 'createScriptURL'>): TrustedScriptURL;
    }
```

Reply:
Thanks, fixed. I originally did this to separate FullTypedPolicy from the rest of the interfaces since it is not part of the spec. But @engelsdamien pointed out to me that the correct approach is to move these helpers out of `declare global`, which I did.